### PR TITLE
Style entities decoder

### PR DESCRIPTION
**Pull request by LucaDevelop**
Hi there,
I've created this class to read the text (or the caption) and the entities from message object and return the text with inline entities (HTML, MarkdownV2 or Markdown) that replicate the original styles but with the possibility to manipulate it afterwards.
I've used it in several bots of mine alongside your library and I've decide to add it directly in the Message class.

This is the repo of my class: https://github.com/LucaDevelop/telegram-entities-decoder

I hope it can be useful.

### CODE REVIEW

File: src/Telegram/Types/Message.php [hunk 1]
```diff
@@ -417,4 +418,16 @@ protected function mapSubObjects(string $key, array $data): TelegramTypes
         // Return always null if none of the objects above matches
         return parent::mapSubObjects($key, $data);
     }
+
+    /**
+     * Decode style entities from Telegram bot messages (bold, italic, etc.) in text with inline entities that duplicate (when possible) the exact style the message had originally when was sended to the bot.
```
Comment (unreal4u) [inline src/Telegram/Types/Message.php:423]: PHPDocumentor require us to come up with a title that explains in less than 80 chars what a method will do.

In the long description (with 2 enters after the title) you can go as long as you want, while always adhering to the 120 chars line length :)

File: src/Telegram/Types/Message.php [hunk 2]
```diff
@@ -417,4 +418,16 @@ protected function mapSubObjects(string $key, array $data): TelegramTypes
...
+     *
+     * @param string $style
```
Comment (unreal4u) [inline src/Telegram/Types/Message.php:425]: As for comments, this is the place to talk about accepted values. 

You can do something like `@param string $style Use 'HTML', 'MarkdownV2' or 'Markdown'`

File: src/Telegram/Types/Message.php [hunk 3]
```diff
@@ -417,4 +418,16 @@ protected function mapSubObjects(string $key, array $data): TelegramTypes
...
+     * @return string
+     */
+    public function decodeEntities($style = 'HTML')
+    {
+        $entitiesDecoder = new EntityDecoder($style);
```
Comment (unreal4u) [inline src/Telegram/Types/Message.php:430]: What happens if the passed style is invalid? Does it throw an exception? If so, add a `@throws` comment above or perform some kind of validation here. 

Ideally, you library should handle this but I saw you don't throw anything on the constructor here: https://github.com/LucaDevelop/telegram-entities-decoder/blob/master/src/EntityDecoder.php#L30

File: src/Telegram/Types/Message.php [hunk 4]
```diff
@@ -417,4 +418,16 @@ protected function mapSubObjects(string $key, array $data): TelegramTypes
...
         return parent::mapSubObjects($key, $data);
     }
+
+    /**
+     * Decode style entities from Telegram bot messages (bold, italic, etc.) in text with inline entities that duplicate (when possible) the exact style the message had originally when was sended to the bot.
+     *
+     * @param string $style
+     * @return string
+     */
+    public function decodeEntities($style = 'HTML')
```
Comment (unreal4u) [inline src/Telegram/Types/Message.php:428]: Don't forget to add the return type `: string` to this method :)

**Review by unreal4u — CHANGES_REQUESTED** [review]
hi!

Thanks for submitting this PR! 

While I am a bit reticent to add methods to a `Type` (which comes from Telegram themselves) I understand there is no other place where we could do this type of stuff and there would need to be some major refactoring in order for us to be able to do this in any other way. 

Can you pick up the comments I left so that I can approve it? Thanks!

**Comment by LucaDevelop** [comment]
Hi,
thanks for considering my PR. I'm on it with the changes you've requested.

**Comment by unreal4u** [comment]
Sorry it took so long @LucaDevelop but I didn't receive ANY notification of your changes! Just happened to stumble across it now and saw that you submitted all changes, I'll review my notification settings so it doesn't happen again!

**Pull request merged**